doc/adrs/031_separate_metadata_from_representations.adoc

```diff
@@ -0,0 +1,101 @@
+= ADR-031 - Separate metadata representations
+
+== Context
+
+Our representations contain two types of information, some of them are related to the content of the representation and other to its identity.
+When we are manipulating representations, for example to display them in the explorer, we need to manipulate properties related to their identity.
+When we are refreshing a representation, we are mostly updating properties related to its content.
+
+We will split those two set of properties into different objects.
+
+
+== Decision
+
+=== GraphQL API change
+
+The representation metadata will also appear in the GraphQL schema used by Sirius Components in order to be able to request only identity properties for a representation.
+For that, we will introduce an interface named `IRepresentationMetadata` with the following content in the schema:
+
+```
+interface RepresentationMetadata {
+  id: ID!
+  descriptionId: ID!
```

Comment:
The kind is removed here in purpose since it can be found by the real type of the object returned. RepresentationMetada being an interface it would be odd to return a DiagramMetadata object with a type `Diagram` and a FormMetadata with a type `Form` etc. It serves no real purpose since it duplicates the type of the object.